src/deserialize.rs

```diff
@@ -0,0 +1,405 @@
+//! `feature = "deserialize"` Serialize and deserialize interfaces.
+//!
+//! This API is only available when SQLite is compiled with `SQLITE_ENABLE_DESERIALIZE`.
+//! These functions create and read a serialized file in-memory, useful on platforms without a real file system
+//! like WebAssembly or Cloud Functions.
+//!
+//! ```
+//! # use rusqlite::{Result, Connection, DatabaseName, NO_PARAMS};
+//! let db = Connection::open_in_memory().unwrap();
```

Comment:
Please reformat these example lines to be a little cleaner. You should also use `?`, like the following:
```rust
//! # fn main() -> Result<()> {
//! // example here
//! # Ok(())
//! # }
```